aiida_codtools/cli/misc/cif_import.py

```diff
@@ -0,0 +1,142 @@
+
+
+@command()
+@options.group(help='Group in which to store the raw imported CifData nodes')
+@click.option(
+    '-d', '--database', type=click.Choice(['cod', 'icsd', 'mpds']), default='cod', show_default=True,
+    help='Select the database to import from'
+)
+@click.option(
+    '-m', '--max-entries', type=click.INT, default=None, show_default=True, required=False,
+    help='Maximum number of entries to import'
+)
+@click.option(
+    '-x', '--number-species', type=click.INT, default=None, show_default=True,
+    help='Import only cif files with this number of different species'
+)
+@click.option(
+    '-s', '--importer-server', type=click.STRING, required=False,
+    help='Optional server address thats hosts the database'
+)
+@click.option(
+    '-h', '--importer-db-host', type=click.STRING, required=False,
+    help='Optional hostname for the database'
+)
+@click.option(
+    '-n', '--importer-db-name', type=click.STRING, required=False,
+    help='Optional name for the database'
+)
+@click.option(
+    '-p', '--importer-db-password', type=click.STRING, required=False,
+    help='Optional password for the database'
+)
+@click.option(
+    '-u', '--importer-api-url', type=click.STRING, required=False,
+    help='Optional API url for the database'
+)
+@click.option(
+    '-a', '--importer-api-key', type=click.STRING, required=False,
+    help='Optional API key for the database'
+)
+def launch(group, database, max_entries, number_species, importer_server, importer_db_host,
+    importer_db_name, importer_db_password, importer_api_url, importer_api_key):
+    """
+    Import cif files from various structural databases, store them as CifData nodes and add them to a Group.
+    Note that to determine which cif files are already contained within the Group in order to avoid duplication,
+    the attribute 'source.id' of the CifData is compared to the source id of the imported cif entry. Since there
+    is no guarantee that these id's do not overlap between different structural databases and we do not check
+    explicitly for the database, it is advised to use separate groups for different structural databases.
+    """
+    from CifFile.StarFile import StarError
+    from urllib2 import HTTPError
+    from aiida.tools.dbimporters import DbImporterFactory
+
+    importer_parameters = {}
+    query_parameters = {}
+
+    if importer_server is not None:
+        importer_parameters['server'] = importer_server
+
+    if importer_db_host is not None:
+        importer_parameters['host'] = importer_db_host
+
+    if importer_db_name is not None:
+        importer_parameters['db'] = importer_db_name
+
+    if importer_db_password is not None:
+        importer_parameters['passwd'] = importer_db_password
+
+    if importer_api_url is not None:
+        importer_parameters['url'] = importer_api_url
+
+    if importer_api_key is not None:
+        importer_parameters['api_key'] = importer_api_key
+
+    if number_species is not None:
```

Comment:
Maybe add something like:
if number_species is None and database=='mpds' then raise an error